docs/guidance/volcano-integration.md

```diff
@@ -0,0 +1,395 @@
+
+### Install KubeRay Operator with Batch Scheduling
+
+Deploy the KubeRay Operator with the `--enable-batch-scheduler` flag to enable Volcano batch scheduling support.
+
+When installing via Helm, you can set the following in your `values.yaml` file:
+
+```
+batchScheduler:
+    enabled: true
+```
+
+Or on the command line:
+
+```
+# helm install kuberay-operator --set batchScheduler.enabled=true
+```
+
+## Run Ray Cluster with Volcano scheduler
+
+Add the `ray.io/scheduler-name: volcano` label to your RayCluster CR to submit the cluster pods to Volcano for scheduling.
+
+Example:
+
+```
+apiVersion: ray.io/v1alpha1
+kind: RayCluster
+metadata:
+  name: test-cluster
+  labels:
+    ray.io/scheduler-name: volcano
+    volcano.sh/queue-name: kuberay-test-queue
+spec:
+  rayVersion: '2.1.0'
+  headGroupSpec:
+    serviceType: ClusterIP
+    rayStartParams:
+      block: 'true'
+    replicas: 1
+    template:
+      spec:
+        containers:
+        - name: ray-head
+          image: rayproject/ray:2.1.0
+          resources:
+            limits:
+              cpu: "1"
+              memory: "2G"
+            requests:
+              cpu: "1"
+              memory: "2G"
+  workerGroupSpecs: []
+```
+
+The following labels can also be provided in the RayCluster metadata:
+
+- `ray.io/priority-class-name`: the cluster priority class as defined by Kubernetes [here](https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/#priorityclass).
+- `volcano.sh/queue-name`: the Volcano [queue](https://volcano.sh/en/docs/queue/) name the cluster will be submitted to.
+
+If autoscaling is enabled, `minReplicas` will be used for gang scheduling, otherwise the desired `replicas` will be used.
+
+### Example: Gang scheduling
+
+In this example, we'll walk through how gang scheduling works with Volcano and KubeRay.
+
+First, let's create a queue with a capacity of 4 CPUs and 4Gi of RAM:
+
+```
+$ kubectl create -f - <<EOF
+apiVersion: scheduling.volcano.sh/v1beta1
+kind: Queue
+metadata:
+  name: kuberay-test-queue
+spec:
+  weight: 1
+  capability:
+    cpu: 4
+    memory: 4Gi
+EOF
+```
+
+Next we'll create a RayCluster with a head node and two workers, each requesting 1 CPU and 1Gi of RAM each, for a total of 3 CPU and 3Gi of RAM:
+
+```
+$ kubectl create -f - <<EOF
+apiVersion: ray.io/v1alpha1
+kind: RayCluster
+metadata:
+  name: test-cluster-0
+  labels:
+    ray.io/scheduler-name: volcano
+    volcano.sh/queue-name: kuberay-test-queue
+spec:
+  rayVersion: '2.1.0'
+  headGroupSpec:
+    serviceType: ClusterIP
+    rayStartParams:
+      block: 'true'
+    replicas: 1
+    template:
+      spec:
+        containers:
+        - name: ray-head
+          image: rayproject/ray:2.1.0
+          resources:
+            limits:
+              cpu: "1"
+              memory: "1Gi"
+            requests:
+              cpu: "1"
+              memory: "1Gi"
```

Comment:
It's dangerous to allocate less than 2Gi memory for the Ray head,
even for an example that depend on Ray's operation at all.
```suggestion
              memory: "2Gi"
```
It's better to present a 2Gi head and one 1 Gi worker,
instead of a 1Gi head and two 1Gi workers.